### PR TITLE
Fix Intune role assignment and cloud login variable

### PR DESCRIPTION
**Pull request by FabienTschanz**
#### Pull Request (PR) description
This PR fixes three things: 
* First, it improves the overall output of `IntuneRoleAssignment` and some copy-pasted variables
* Secondly, it fixes the incorrect variable `$Global:MSCloudLoginAssistant` to `$Global:MSCloudLoginConnectionProfile`
* Thirdly, it adds the `AccessTokens` property to the output of the Get method

#### This Pull Request (PR) fixes the following issues
None.

#### Task list
- [x] Added an entry to the change log under the Unreleased section of the file CHANGELOG.md.
      Entry should say what was changed and how that affects users (if applicable), and
      reference the issue being resolved (if applicable).
- [ ] Resource parameter descriptions added/updated in the schema.mof.
- [ ] Resource documentation added/updated in README.md.
- [ ] Resource settings.json file contains all required permissions.
- [ ] Examples appropriately added/updated.
- [ ] Unit tests added/updated.
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

### CODE REVIEW

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneRoleAssignment/MSFT_IntuneRoleAssignment.psm1 [hunk 1]
```diff
@@ -46,7 +46,7 @@ function Get-TargetResource
 
         [Parameter()]
         [System.String]
-        [ValidateSet('Absent', 'Present')]
+        [ValIdateSet('Absent', 'Present')]
```
Comment (ricmestre) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneRoleAssignment/MSFT_IntuneRoleAssignment.psm1]: Please change this :)

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneRoleAssignment/MSFT_IntuneRoleAssignment.psm1 [hunk 2]
```diff
@@ -259,7 +234,7 @@ function Set-TargetResource
...
```
Comment (ricmestre) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneRoleAssignment/MSFT_IntuneRoleAssignment.psm1]: Also here

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneRoleAssignment/MSFT_IntuneRoleAssignment.psm1 [hunk 3]
```diff
@@ -472,7 +437,7 @@ function Test-TargetResource
...
```
Comment (ricmestre) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneRoleAssignment/MSFT_IntuneRoleAssignment.psm1]: And here

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneRoleAssignment/MSFT_IntuneRoleAssignment.psm1 [hunk 4]
```diff
@@ -730,7 +695,7 @@ function Export-TargetResource
     }
     catch
     {
-        if ($_.Exception -like '*401*' -or $_.ErrorDetails.Message -like "*`"ErrorCode`":`"Forbidden`"*" -or `
+        if ($_.Exception -like '*401*' -or $_.ErrorDetails.Message -like "*`"ErrorCode`":`"ForbIdden`"*" -or `
```
Comment (ricmestre) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneRoleAssignment/MSFT_IntuneRoleAssignment.psm1]: And here

**Comment by ricmestre** [comment]
Apart from the wrong casing it looks good to me, fortunately (or rather unfortunate) the reason why the resources didn't fail because of the MSCloudLoginAssistant variable is due to Graph not complaining about the URIs used anywhere as long as they start with the API name (beta vs v1.0) which they did because the MSCloudLoginAssistant variable was always null.

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneRoleAssignment/MSFT_IntuneRoleAssignment.psm1 (hunk 4 above)
Comment (FabienTschanz) [reply Modules/Microsoft365DSC/DSCResources/MSFT_IntuneRoleAssignment/MSFT_IntuneRoleAssignment.psm1]: That's what I get from Ctrl + H 😆

**Comment by FabienTschanz** [comment]
@ricmestre True dat, I only stumbled upon it because the `IntuneRoleAssignment` had a direct relation to such a URL which was failing. Thanks for the review, I updated the casing.

**Review by NikCharlebois — APPROVED** [review]

**Pull request merged**